### DISCUSSION TITLE
Passing data from controller / showing a dynamic subset?

### DISCUSSION THREAD

**Discussion by mallanza** [rappasoft/laravel-livewire-tables#1626] [Q&A] [answered]
I am relatively new to Laravel and don't have much XP working with Livewire.
So I am wondering how do I open a table (I wan to use Modal) and pass a list of ID's to display?
For example, I have another table "Roles' that I can use to use Role model and in a column display number of users for each role. I want to click on that number and open Users table and only preselect those user ID's that I've passed.

**Comment by lrljoe** [comment, marked as answer]
This is quite a complex question that you've asked here to be honest, as you're basically asking for a solution to role-based access control GUI.

Taking it a step at a time, you have a "Roles" table.

I'm assuming presence of Eloquent relations, so you can use the relationship to the "Users" to get a "count" back, and are doing that?

You can then add a label/view/button Column, and use that to open a Modal, passing relevant parameters (for example, the "role id".

When it comes to then opening the "Users" table, there are various options, and you can expose different properties, and use them on mount, you can then use these in the builder() as you need (to constrain the relation for example).

It's a **very** complex question to be honest, and there are so many variables that it's hard to give a solution easily without more detail!  In short - yes, it's very possible, and I use the tables package in conjunction with modals etc, without issue.

For example - are you using an RBAC package?

**Reply by mallanza** [reply]
I will look into this more deeply, and try passing an array with IDs into the table, using mount() in combination with builder() to parse IDs and then rendering the table accordingly.
PS- I am using spatie/laravel-permission for roles/permissions
